### PR TITLE
P#65163 Detail page leads to error when it has a parent page

### PR DESCRIPTION
**Pull request by hungnc89**

### CODE REVIEW

File: plugin/EstateList.php [hunk 1]
```diff
@@ -99,6 +102,12 @@ class EstateList
 	/** @var EstateDetailUrl */
 	private $_pLanguageSwitcher;
 
+	/** @var Redirector */
+	private $_redirectIfOldUrl;
+
+	/** @var bool */
+	private $_isOverride_DataView = false;
```
Comment (jayay) [inline plugin/EstateList.php:109]: I was hoping this wouldn't be necessary, especially it only exists for the tests

File: plugin/Utility/Redirector.php [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+		$pageName = $this->_wpPageWrapper->getPageUriByPageId($pageId);
+		preg_match('/^(' . preg_quote($pageName, '/') .')\/([0-9]+)(-([^$]+))?\/?$/', $uri, $matches);
+		if (empty($matches[2])) { //Check pass rule and has Unique ID
+			return true;
+		}
+		if ($fullLink != $currentLink) {
+			$this->_wpRedirectWrapper->redirect($fullLink);
+		}
+
+		return true;
```
Comment (jayay) [inline plugin/Utility/Redirector.php:39]: Is the return value necessary?

**Review by jayay — COMMENTED** [review]
I had two things annotated

**Review by jayay — DISMISSED** [review]

**Comment by michaelarnold-dev** [comment]
@hungnc89 
Version 2.21.7-18-gfcdf53e4 imstalled on https://wpptest.onofficeweb.com/:

Some estate detail views do not load. It comes "Page not available". See video. Cause unclear.
--> All detail views displayed in the list should open correctly

https://user-images.githubusercontent.com/77264788/160138457-45be570e-879f-4322-9d02-34955eb42ba5.mp4

.

**Comment by LongTrong-exe** [comment]
@michaelarnold-dev oh sorry, let me update this brand. your bug you said is a old feature, It has been updated in the new version.

**Review by jayay — APPROVED** [review]

**Review by jayay — DISMISSED** [review]

**Review by jmaas-onoffice — CHANGES_REQUESTED** [review]
The bug with the parent page is fixed, thanks!

Unfortunately there still is a bug in [version 2.22.5-21-ge0af7dd8]():

When I "break" the title, the redirection fails with the error message `Cannot modify header information - headers already sent`.

https://user-images.githubusercontent.com/90763364/161999964-d92a8592-b117-42b9-a0c3-9c9328518178.mp4

I have noticed that this error does not occur with the theme "Twenty Twenty-Two". It does occurr with the theme "Neve". I have not tested other themes. I have set up "Neve" on https://2.wptest.onofficeweb.com/, and you can use that server for testing.

Could you please fix this?):

**Comment by LongTrong-exe** [comment]
@jmaas-onoffice I fixed the bug, this's my test video:

https://user-images.githubusercontent.com/101325858/162874998-950e1328-b65b-420e-8a26-39379f5e84a1.mp4

**Review by jmaas-onoffice — APPROVED** [review]

**Pull request merged**